src/zenml/steps/entrypoint_function_utils.py

```diff
@@ -235,17 +235,23 @@ def _validate_input_value(
             parameter: The function parameter for which the value was provided.
             value: The input value.
         """
-        config_dict = ConfigDict(arbitrary_types_allowed=False)
-
-        # Create a pydantic model with just a single required field with the
-        # type annotation of the parameter to verify the input type including
-        # pydantics type coercion
-        validation_model_class = create_model(
-            "input_validation_model",
-            __config__=config_dict,
-            value=(parameter.annotation, ...),
-        )
-        validation_model_class(value=value)
+        annotation = parameter.annotation
+
+        # Use Pydantic for all types to take advantage of its coercion abilities
+        try:
+            config_dict = ConfigDict(arbitrary_types_allowed=True)
+            validation_model_class = create_model(
+                "input_validation_model",
+                __config__=type("Config", (), config_dict),
+                value=(annotation, ...),
+            )
+            validation_model_class(value=value)
+        except Exception:
```

Comment:
Should this be 
```python
try:
  ...
except ValidationError:
  raise
except Exception:
  ...
```
instead? In case pydantic fails with a regular validation error, we actually want to raise that because it means the types don't match right?